app/request_handler.py

```diff
@@ -225,17 +225,25 @@ def writeEventRecord(eventObj):
     )
 
 def getEventfulEventObj(event):
-    locLat = float(event['latitude'])
-    locLng = float(event['longitude'])
-
-    yelpId = _guessYelpId(event['venue_name'], locLat, locLng)
-
-    # Set local timezone
-    startTime, endTime = events.getTimesWithTZ(event["start_time"], event["stop_time"], locLat, locLng)
+    mappingLat, mappingLng = getEventfulMapping(event)
+    if (mappingLat != None and mappingLng != None):
+        yelpId = _guessYelpId(event['venue_name'], mappingLat, mappingLng)
+        # Set local timezone
+        startTime, endTime = events.getTimesWithTZ(event["start_time"], event["stop_time"], mappingLat, mappingLng)
+
+        if yelpId:
+            eventObj = representation.eventRecord(yelpId, mappingLat, mappingLng, event['title'], str(startTime), str(endTime), event['url'])
+            return eventObj
+
+def getEventfulMapping(event):
+    venueAddress = event["venue_address"]
+    venueName = event["venue_name"]
+    mapping = search._getAddressIdentifiers(venueAddress) if venueAddress else search._findPlaceInRange(venueName, event["latitude"], event["longitude"])
```

Comment:
Technical debt: rename and document `_getAddressIdentifiers` and `_findPlaceInRange` to reflect what they do.

```python
if venueAddress:
    # The Brunswick, Waterloo Street, Hove, BN2 1AA
    mappping = … # find a lat long from the venue address using Google Places Geocode
else:
    # The Brunswick, -0.034, 51.345
    mapping = … # find an accurate lat long from a text search and a rough lat long taken from eventful. The eventful lat long are not as reliable as Google Places.